File: Frontend/src/components/utils/validators.js
export function validateEmail(email) {
  const re = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  return re.test(email);
}

export function validatePassword(password) {
  return password && password.length >= 6;
}

export function validateRequired(value) {
  return value !== undefined && value !== null && value !== '';
}

export function validateNumber(value) {
  const n = Number(value);
  return !isNaN(n) && n >= 0;
}

export function validateLoginForm({ email, password }) {
  const errors = {};
  if (!validateEmail(email)) {
    errors.email = 'Please enter a valid email address';
  }
  if (!validatePassword(password)) {
    errors.password = 'Password must be at least 6 characters';
  }
  return errors;
}

File: backend/utils/validators.py
import re

def validate_email(email):
    """Validate email format"""
    pattern = r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$'
    return re.match(pattern, email) is not None

def validate_password(password):
    """Validate password strength (min 6 characters)"""
    return len(password) >= 6

def validate_login_data(email, password):
    """Validate login data"""
    errors = []
    
    if not email:
        errors.append("Email is required")
    elif not validate_email(email):
        errors.append("Invalid email format")
    
    if not password:
        errors.append("Password is required")
    elif not validate_password(password):
        errors.append("Password must be at least 6 characters")
    
    return errors